internal/controller/gateway.go

```diff
@@ -366,6 +366,23 @@ func (c *GatewayController) bspToFilterAPIBackendAuth(ctx context.Context, names
 		return &filterapi.BackendAuth{
 			AzureAuth: &filterapi.AzureAuth{AccessToken: azureAccessToken},
 		}, nil
+	case aigv1a1.BackendSecurityPolicyTypeGCPCredentials:
+		gcpCreds := backendSecurityPolicy.Spec.GCPCredentials
+		if gcpCreds == nil {
```

Comment:
could u reuse validateGCPCredentialsParams here? or if its already called would this be needed?

Reply:
we don't need to validate params inside GCPCredentials here as the struct is directly coming from a CR. The CEL validations on the CR itself would have ensured the gcpCreds are valid. cel-validation are defined here (https://github.com/envoyproxy/ai-gateway/blob/58a01871ba07fd61001c314616251197edf720f5/api/v1alpha1/api.go#L625-L640)